system-addon/common/PrerenderData.jsm

```diff
@@ -1,32 +1,80 @@
-const prefConfig = {
-  // Prefs listed with "invalidates: true" will prevent the prerendered version
+class _PrerenderData {
+  constructor(options) {
+    this.initialPrefs = options.initialPrefs;
+    this.initialSections = options.initialSections;
+    this._setValidation(options.validation);
+  }
+
+  get validation() {
+    return this._validation;
+  }
+
+  set validation(value) {
+    this._setValidation(value);
```

Comment:
Not sure if we really need these getter/setters. But I suppose they're fine.

Reply:
I figured this would be easier than having cacheing for `.invalidatablePrefs` since it's computed from `.validation`, but it's not strictly necessary if we don't care about modifying `.validation` after the constructing the instance